Checkout service does not build with dotnet 1.1.2 sdk

Hi,

There seems to be some issue with building the CheckoutService using 1.1.2-sdk. I didn't spend enough time checking what is going on, usually something in csproj is at fault. Below are the steps I followed.

I have the latest master cloned (hash 81af523987352da11f1ecd7444fa3617d80f84a4).
From clone directory I tried to do docker-compose -f .\docker-compose.dev.yml build, and I noticed the CheckoutService does not build.

I moved to the checkoutService directory and ran docker build -t checkout:latest .
Still the build broke.

```
 Installing MongoDB.Driver.Core 2.4.3.
  Generating MSBuild file /app/obj/CheckoutService.csproj.nuget.g.props.
  Generating MSBuild file /app/obj/CheckoutService.csproj.nuget.g.targets.
  Restore completed in 5.48 sec for /app/CheckoutService.csproj.
 ---> 3c29412175d6
Removing intermediate container 413264aef346
Step 5/7 : COPY . ./
 ---> 505be56cc6c3
Removing intermediate container 1c17d5ea9ecd
Step 6/7 : RUN dotnet publish -c Release -o out
 ---> Running in b88f2cc309c3
Microsoft (R) Build Engine version 15.3.409.57025 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

/usr/share/dotnet/sdk/1.1.0/Sdks/Microsoft.NET.Sdk/build/Microsoft.PackageDependencyResolution.targets(308,5): error : Assets file
'/app/c:/prj/react/StickerApp/checkoutService/obj/project.assets.json' not found. Run a NuGet package restore to generate this file
. [/app/CheckoutService.csproj]
/usr/share/dotnet/sdk/1.1.0/Sdks/Microsoft.NET.Sdk/build/Microsoft.PackageDependencyResolution.targets(157,5): error : Assets file
'/app/c:/prj/react/StickerApp/checkoutService/obj/project.assets.json' not found. Run a NuGet package restore to generate this file
. [/app/CheckoutService.csproj]
/tmp/.NETCoreApp,Version=v1.1.AssemblyAttributes.cs(4,20): error CS0400: The type or namespace name 'System' could not be found in
the global namespace (are you missing an assembly reference?) [/app/CheckoutService.csproj]
obj/Release/netcoreapp1.1/CheckoutService.AssemblyInfo.cs(10,12): error CS0246: The type or namespace name 'System' could not be fo
und (are you missing a using directive or an assembly reference?) [/app/CheckoutService.csproj]
obj/Release/netcoreapp1.1/CheckoutService.AssemblyInfo.cs(11,12): error CS0246: The type or namespace name 'System' could not be fo
und (are you missing a using directive or an assembly reference?) [/app/CheckoutService.csproj]
obj/Release/netcoreapp1.1/CheckoutService.AssemblyInfo.cs(12,12): error CS0246: The type or namespace name 'System' could not be fo
und (are you missing a using directive or an assembly reference?) [/app/CheckoutService.csproj]
obj/Release/netcoreapp1.1/CheckoutService.AssemblyInfo.cs(13,12): error CS0246: The type or namespace name 'System' could not be fo
und (are you missing a using directive or an assembly reference?) [/app/CheckoutService.csproj]
obj/Release/netcoreapp1.1/CheckoutService.AssemblyInfo.cs(14,12): error CS0246: The type or namespace name 'System' could not be fo
und (are you missing a using directive or an assembly reference?) [/app/CheckoutService.csproj]
```
Restore happened with no issues, however publish fails for some reason.

I changed the base image to microsoft/dotnet:2.0-sdk, ran the docker build command. Everything now works fine.

If you want to keep the service at 1.1.2 then can you please see why it fails?

I'm off playing with the demo ;)

--- a/checkoutService/Dockerfile
+++ b/checkoutService/Dockerfile
@@ -1,13 +1,11 @@
-FROM microsoft/dotnet:1.1.2-sdk
-WORKDIR /app
-
-# copy csproj and restore as distinct layers
-COPY CheckoutService.csproj ./
+FROM microsoft/aspnetcore-build:1.1.4 AS builder
+WORKDIR /build
+COPY . .
 RUN dotnet restore
+RUN dotnet publish -c Release -o /publish-output
 
-# copy and build everything else
-COPY . ./
-RUN dotnet publish -c Release -o out
-
-CMD ["dotnet", "out/CheckoutService.dll", "--server.urls", "http://*:5000"]
+FROM microsoft/aspnetcore:1.1.4
+WORKDIR /app
+COPY --from=builder /publish-output .
+CMD ["dotnet", "CheckoutService.dll"]
 

--- a/docker-compose.dev.yml
+++ b/docker-compose.dev.yml
@@ -48,6 +48,7 @@ services:
     - 5000:5000
     environment:
       ASPNETCORE_ENVIRONMENT: dev
+      ASPNETCORE_URLS: "http://*:5000"
     links:
     - mongo
 

